mempool/reactor.go

```diff
@@ -19,17 +20,21 @@ import (
 // peers you received it from.
 type Reactor struct {
 	p2p.BaseReactor
-	config  *cfg.MempoolConfig
-	mempool *CListMempool
-	ids     *mempoolIDs
+	config   *cfg.MempoolConfig
+	mempool  *CListMempool
+	ids      *mempoolIDs
+	waitSync *atomic.Bool
 }
 
 // NewReactor returns a new Reactor with the given config and mempool.
-func NewReactor(config *cfg.MempoolConfig, mempool *CListMempool) *Reactor {
+func NewReactor(config *cfg.MempoolConfig, mempool *CListMempool, waitSync bool) *Reactor {
```

Comment:
Is `waitSync` ever initialized to true? I mean, does its initial value varies depending on the scenario?

Reply:
I'm not sure I understand the question. `waitSync` is initialized with the same value as the `waitSync` variable in the consensus reactor. So it's initially true when `stateSync || blockSync` is true, as defined in the node. In both reactors, if `waitSync` is initially true, it will be set to false by the methods `EnableIncomingTxs` or `SwitchToConsensus`.